mypy_django_plugin/transformers/settings.py

```diff
@@ -20,6 +27,28 @@ def get_user_model_hook(ctx: FunctionContext, django_context: DjangoContext) ->
     return TypeType(Instance(model_info, []))
 
 
+def transform_get_user_model_hook(ctx: DynamicClassDefContext, django_context: DjangoContext) -> None:
+    auth_user_model = django_context.settings.AUTH_USER_MODEL
+    try:
+        model_cls = django_context.apps_registry.get_model(auth_user_model)
+        model_cls_fullname = helpers.get_class_fullname(model_cls)
+    except LookupError:
+        model_cls_fullname = fullnames.ABSTRACT_BASE_USER_MODEL_FULLNAME
```

Comment:
If we have my lazy reference resolver in #1719 we wouldn't need to use the runtime model here:

```suggestion
    auth_user_model = django_context.settings.AUTH_USER_MODEL
    model_cls_fullname = helpers.resolve_lazy_reference(auth_user_model, ...)
    if model_cls_fullname is None:
        model_cls_fullname = fullnames.ABSTRACT_BASE_USER_MODEL_FULLNAME
```

We might want to use the fallback only on last iteration then